### DISCUSSION TITLE
Need help quick regarding MSIL

### DISCUSSION THREAD

**Discussion by xx7Ahmed7xx** [icsharpcode/ILSpy#2514] [Help]
How exactly to convert MSIL that i got using "DecompileWholeModuleAsSingleFile()" into C# code ??

**Comment by siegfriedpammer** [comment]
Sorry, I don't exactly understand the question: `DecompileWholeModuleAsSingleFile` already produces C# code, it's just the abstract syntax tree representation and not a string. If you want a string, you can use the following (untested) code:

```csharp
// produce AST
var syntaxTree = decompiler.DecompileWholeModuleAsSingleFile();

// produce string
StringWriter output = new StringWriter();
var visitor = new CSharpOutputVisitor(output, FormattingOptionsFactory.CreateSharpDevelop());
syntaxTree.AcceptVisitor(visitor);

// write to file
string fileName = Path.GetTempFileName();
File.WriteAllText(fileName, output.ToString());
```

Hope this helps!
